src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -159,6 +181,87 @@ private string GetRootPathWithGuid_NoLock()
             return new MetadataAsSourceFile(fileInfo.TemporaryFilePath, navigateLocation, documentName, documentTooltip);
         }
 
+        private async Task<Document> DecompileSymbolAsync(Document temporaryDocument, ISymbol symbol, CancellationToken cancellationToken)
+        {
+            // Get the name of the type the symbol is in:
+            var containingOrThis = symbol.GetContainingTypeOrThis();
+            var total = GetFullReflectionName(containingOrThis);
+
+            var model = await temporaryDocument.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            // TODO : retrieve path to actual assembly instead of reference assembly:
+            var reference = model.Compilation.GetMetadataReference(symbol.ContainingAssembly);
+            // Load the assembly.
+            var ad = AssemblyDefinition.ReadAssembly(reference.Display, new ReaderParameters() { AssemblyResolver = new RoslynAssemblyResolver(model.Compilation) });
+
+            // Initialize a decompiler with default settings.
+            CSharpDecompiler decompiler = new CSharpDecompiler(ad.MainModule, new DecompilerSettings());
+            var fullTypeName = new FullTypeName(total);
+
+            // Try to decompile
+            // If it fails, display stack trace and assembly file name.
+            string text;
+            try
+            {
+                text = decompiler.DecompileTypeAsString(fullTypeName);
+            }
+            catch (Exception ex)
+            {
+                text = ad.MainModule.FileName + "\n" + ex.ToString();
```

Comment:
Consider adding some text at the start that says something like "We failed to decompile the file. You might want to turn this off under Tools Options, etc."

Reply:
:arrow_right: No longer applicable